internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -550,16 +665,22 @@ func (s *service) pathFromToken(ctx context.Context, token string) (string, erro
 			},
 		},
 	)
-	if err != nil {
-		return "", err
+	switch {
+	case err != nil:
+		return "", nil, nil, err
+	case publicShareResponse.Status.Code != rpc.Code_CODE_OK:
+		return "", nil, publicShareResponse.Status, nil
```

Comment:
Returning the status only when it's not OK seems to make it a bit inconsistent for me. Not a big deal but I'd prefer if we always return the status and check if it's OK or not in the caller.

Reply:
We are doing the Status.Code checks here to determine if we can continue or if we need to return the Status so the caller can decide what to do. `resolveToken()` is called in three places and I think this method does exactly what you intend: resolve the token but give me the error or status code if something goes wrong.